packages/liquidator/src/liquidator.js

```diff
@@ -250,52 +273,116 @@ class Liquidator {
         parseInt(currentBlockTime) + this.liquidationDeadline
       );
 
-      // Simple version of inventory management: simulate the transaction and assume that if it fails, the caller didn't have enough collateral.
-      let liquidationResult, gasEstimation;
-      try {
-        [liquidationResult, gasEstimation] = await Promise.all([
-          liquidation.call({ from: this.account }),
-          liquidation.estimateGas({ from: this.account })
+      const syntheticTokenBalance = await this.syntheticToken.balanceOf(this.account);
+      const notEnoughTokens = syntheticTokenBalance.lt(tokensToLiquidate);
+
+      // Attempts to swap some capital for tokenCurrency
+      // Mutates the state of notEnoughCollateral in this logical scope
+      if (notEnoughTokens) {
+        // reserveWeiNeeded is a reverse calculation to estimate how much capital we need to get the amount of `tokensToLiquidate`
+        // While oneGweiReturn is used as a reference point to determine slippage
+        const [reserveWeiNeeded, oneGweiReturn] = await Promise.all([
+          this.oneInchClient.getExpectedReturn({
+            fromToken: this.syntheticToken.address,
+            toToken: this.reserveCurrencyAddress,
+            amountWei: tokensToLiquidate.toString()
+          }),
+          this.oneInchClient.getExpectedReturn({
+            fromToken: this.syntheticToken.address,
+            toToken: this.reserveCurrencyAddress,
+            amountWei: this.toWei("1", "gwei")
+          })
         ]);
-      } catch (error) {
-        this.logger.error({
+
+        this.logger.info({
```

Comment:
plz add an emoji here. all other `info` events have an emoji. you can pick something fitting :)

Reply:
I would also add `reserveWeiNeeded`, `oneGweiReturn`, `tokensToLiquidate` to the Log